tests/30rooms/12thirdpartyinvite.pl

```diff
@@ -0,0 +1,395 @@
+use Crypt::NaCl::Sodium;
+use File::Basename qw( dirname );
+use Protocol::Matrix qw( encode_base64_unpadded );
+use SyTest::Identity::Server;
+
+my @user_preparers = local_user_preparers( 2 );
+
+my $crypto_sign = Crypt::NaCl::Sodium->sign;
+
+my $DIR = dirname( __FILE__ );
+
+test "Can invite existing 3pid",
+   requires => [ @user_preparers ],
+
+   do => sub {
+      my ( $inviter, $invitee ) = @_;
+
+      my $invitee_email = 'marmosets@monkeyworld.org';
+      my $invitee_mxid = $invitee->user_id;
+
+      my $room_id;
+
+      my $stub_id_server = SyTest::Identity::Server->new;
+      $stub_id_server->{bindings}{$invitee_email} = $invitee_mxid;
+      $loop->add( $stub_id_server );
+      require IO::Async::SSL;
```

Comment:
This is subtly hiding in the middle of some code. At least put some blank lines around it; or maybe even hoist it up to the start of the file along with the others.